docs/protocol.html

```diff
@@ -114,6 +114,32 @@
 
 <p>Currently all versions are baselined at 0, as we evolve these APIs we will indicate the format for each version individually.</p>
 
+<h5><a id="api_versions" href="#api_versions">Retrieving Supported API versions</a></h5>
+<p>In order for a client to successfully talk to a broker, it must use requests versions' supported by the broker. Clients may work against
+    multiple broker versions, however to do so the clients need to know what versions of various APIs does a broker supports. Clients may
+    use the supported API versions information to take appropriate actions, like, propagate unsupported api version error to application
+    or choose a version from many API versions supported by client. The following sequence maybe used by a client to obtain supported API
+    versions from a broker.</p>
+<ol>
+    <li>Client sends <code>ApiVersionsRequest</code> to a broker after connection has been established with the broker.</li>
+    <li>If version of <code>ApiVersionsRequest</code> sent by a client is not supported by broker, the broker will respond with an
```

Comment:
This explanation seems premature to add. We only have v0 of ApiVersionsRequest, so we're basically explaining how to handle a currently non-existent use case (and a case we kind of expect to never have to handle since we really don't want a v1 version unless we really can't avoid it). Also, I suspect the right way to handle this case is to start with a v0 request and upgrade to a vN request as supported since it should never be a problem to continue supporting v0 ApiVersionsRequest.

I'd suggest just removing discussion of different ApiVersionsRequest versions for the moment since its a non-issue and clearly we're not all on the same page about the best approach for handling the possibility of multiple versions of that request :)